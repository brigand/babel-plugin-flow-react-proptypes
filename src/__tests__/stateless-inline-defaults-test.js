const babel = require('babel-core');
const content = `
var React = require('react');

export default function Foo({x=1, y='foo'}: {
  x?: number,
  y?: string
}) {
  <div>{x}/{y}</div>
}
`;

it('stateless-inline-defaults', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
