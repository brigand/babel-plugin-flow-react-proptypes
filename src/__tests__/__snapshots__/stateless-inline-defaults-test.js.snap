// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateless-inline-defaults 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = Foo;

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

var React = require('react');

function Foo(_ref) {
  var _ref$x = _ref.x,
      x = _ref$x === void 0 ? 1 : _ref$x,
      _ref$y = _ref.y,
      y = _ref$y === void 0 ? 'foo' : _ref$y;

  /*#__PURE__*/
  React.createElement(\\"div\\", null, x, \\"/\\", y);
}

Foo.propTypes = {
  \\"x\\": _propTypes[\\"default\\"].number,
  \\"y\\": _propTypes[\\"default\\"].string
};"
`;
